.github/workflows/python-ci-full.yml

```diff
@@ -20,8 +20,6 @@ jobs:
             cc: gcc-11
             cxx: g++-11
           - runs-on: macos-12
```

Comment:
> Appears to fix current main CI...unclear if setting gcc was intentional, and we do NOT do that for the wheel builds.

@mlin we can merge this & that's fine but the pre-merge minimal CI doesn't run on MacOS.

Just to verify this works, please _temporarily_ include a `- runs-on: macos-12` line in `python-ci-minimal.yml` and push & make sure it builds OK, then remove it, then merge to `main` with @Shelnutt2 's accpt

Reply:
@mlin I would do Request Changes but I want to be respectful of timezones. Please make this change as requested before merging. Thanks!

Reply:
I'm trying this advice on #1128 ... my own advice may have been misspelt ... 👀